Various fixes & updates

hannamlmv opened the pull request:
See the individual closed PRs on our fork: [here](https://github.com/hannamlmv/tseda/pulls?q=is%3Apr+is%3Aclosed)
Each PR is linked to [closed issues](https://github.com/hannamlmv/tseda/issues?q=is%3Aissue+is%3Aclosed) we have identified, the issues are named based on the page


**Sample sets page:**
- Page size setting is now functional
- Decreased number of samples per page in sidebar quick view table
- Removed the possibility of adding several populations with identical names, and added a warning when the user enters an already existing name
- New, unused colors are assigned randomly to new populations. When all colors are used, repetitions are allowed.


**Individuals page:** 
- Improved on hover information boxes 
- Sample selection fixed
    - Selection fully functional
    - Improved user experience with multiple choice box 
    - Improved description and naming 
- Added filtering on table 
-  Fixed table alignment problem
-  Solved problems with map dimensions
    - Gaps around figure
    - Problems with changing browser window size
- Improved the modification of sample set IDs
    - More helpful descriptions
    - More intuitive UI
- Reorganised sidebar
    - Removing map options
    - Splitting up cards by type of interaction


**Structure page:**
- Placing the 2 plots vertically instead of horizontally
- Making the plots responsive to horizontal resizing of window
- Added warning message when no plot is shown due to less than 2 samples being selected.


**iGNN page:** 
- Making the GNN plot horizontally resizable
- Changed order of sidebar menus to match the order of the plot
- Add the possibility to sort samples in either ascending or descending order
- Changed the default state of sidebar menus (only sample sets quick view table is collapsed)
- Add y-axis labels to all plots on iGNN page so that they are aligned horizontally
- Added warnings for GNN VBar plot & GNN haplotype plot if no samples are selected


**Statistics page:** 
- Set menus to open by default for consistency across pages


**Trees page:** 
- Fix tree size to fit standard window
- Add warnings invalid inputs for position & tree index
- Replace the displayed tree position by a range
- Fix so that the tree index can't increase over the maximum bound
- Add a toggle for searching a tree by position or index


**General:**
 - Removed dark-theme button as it wasn't functional
 - Extended the sleep time for the tests, so that the application has time to render properly

tforest commented:
Very nice work! Thank you :+1: 

The way some parameters are set by default, in datastore.py, are causing the stats to fail by default. The one-way stat should work as soon as one sample set is chosen, however, the default is expecting the sample sets : [0, 1], so it cannot load them if you put any other configuration in the Individuals tabs. Same for the multi-way stat, but it's for the same reason. By default it's expecting the sample set [0,1,2] and indexes [(0,1), (0,2), (1,2)]. 

Some ideas :

When selecting individuals, sort the list so that the selected samples show up first. I did not noticed the second page and was looking for the orangutan at first, when selecting it by it's sample set (5). 

In the trees tab, it would be possible to initialize the first position to the first tree in the tree sequence, by using tree_sequence.first().interval.left or something like that, I guess?

hannamlmv commented:
We have created issues to address your comments. 

To solve the issue with the statistics page, we will set all samples as selected by default (see [issue 67](https://github.com/hannamlmv/tseda/issues/67). Further improvements to the statistics page are coming, see [PR 63](https://github.com/hannamlmv/tseda/pull/63).

We added an issue to fix the sorting of individuals, see [issue 66](https://github.com/hannamlmv/tseda/issues/66). This won't be fixed in this PR, but possible in a future one. 

After discussions with Per, we think that the reason why there is no tree at index 0 is due to the specific test data. This should not be a problem for other datasets. We added an issue for this ([issue 68](https://github.com/hannamlmv/tseda/issues/68) but labeled it as wontfix.

We will solve a few small issues on the Individual's page before this PR is ready to be merged ([67](https://github.com/hannamlmv/tseda/issues/67), [69](https://github.com/hannamlmv/tseda/issues/69) and [70](https://github.com/hannamlmv/tseda/issues/70).
